resotoclient/__init__.py

```diff
@@ -372,20 +403,167 @@ def cli_evaluate(
-            headers={"Content-Type": "text/plain"},
+            headers=headers,
             stream=True,
         )
+        return response
+
+    def cli_execute(
+        self,
+        command: str,
+        graph: Optional[str] = "resoto",
+        section: Optional[str] = "reported",
+        headers: Optional[Dict[str, str]] = None,
+        files: Optional[Files] = None,
+        **env: str,
+    ) -> Iterator[JsValue]:
+        """
+        Execute a CLI command and return the result as a stream of text or JSON objects.
+
+        Binary or multi-part responses will trigger an exception.
+        """
+
+        response = self.cli_execute_raw(
+            command=command,
+            graph=graph,
+            section=section,
+            headers=headers,
+            files=files,
+            **env,
+        )
+
         if response.status_code == 200:
-            return map(lambda l: json_loadb(l), response.iter_lines())
+            content_type = response.headers.get("Content-Type")
+            if content_type == "text/plain":
+                return iter([response.text])
+            elif content_type == "application/json":
+                return iter([response.json()])
+            elif content_type == "application/x-ndjson":
+                return map(lambda line: json_loadb(line), response.iter_lines())
+            else:
+                raise NotImplementedError(
+                    f"Unsupported content type: {content_type}. Use cli_execute_raw or cli_execute_with_callback instead."
+                )
+        else:
+            raise AttributeError(response.text)
+
+    def cli_execute_bytes(
+        self,
+        command: str,
+        graph: Optional[str] = "resoto",
+        section: Optional[str] = "reported",
+        headers: Optional[Dict[str, str]] = None,
+        files: Optional[Files] = None,
+        **env: str,
+    ) -> Iterator[bytes]:
+        """
+        Execute a CLI command and return the result as a stream of binary data.
+
+        Text or multi-part responses will trigger an exception.
+        """
+
+        response = self.cli_execute_raw(
+            command=command,
+            graph=graph,
+            section=section,
+            headers=headers,
+            files=files,
+            **env,
+        )
+
+        if response.status_code == 200:
+            content_type = response.headers.get("Content-Type")
+            if content_type == "application/octet-stream":
+                return response.iter_content(chunk_size=8192)
+            else:
+                raise NotImplementedError(
+                    f"Unsupported content type: {content_type}. Use other cli_execute_raw or cli_execute_with_callback function instead."
+                )
+        else:
+            raise AttributeError(response.text)
+
+    def cli_execute_with_callback(
+        self,
+        command: Command,
+        result_handler: Dict[ContentType, Callable[[Response], None]],
```

Comment:
Maybe simply a function, which is called for every response. The callable itself can decide by content-type or any other criteria.
```suggestion
        result_handler: Callable[[Response], None],
```